File: e2e/selenide-junit5-selenium-grid/src/test/java/context/TaskTestContext.java
package context;

import data.*;
import io.restassured.response.Response;
import lombok.Builder;
import lombok.Data;

import java.util.UUID;

import static org.apache.commons.lang3.RandomStringUtils.secure;

@Builder
@Data
public class TaskTestContext {
    @Builder.Default
    private String id = UUID.randomUUID().toString();

    @Builder.Default
    private String title = secure().nextAlphabetic(12);

    @Builder.Default
    private String description = secure().nextAlphabetic(12);

    @Builder.Default
    private TaskStatus status = TaskStatus.TODO;

    @Builder.Default
    private TaskPriority priority = TaskPriority.MEDIUM;

    private Response response;

    public void setResponse(Response response) {
        this.response = response;
        this.id = response.jsonPath().get("id");
    }

    public TaskData createTaskData() {
        return TaskData.builder()
                .title(title)
                .description(description)
                .status(status)
                .priority(priority)
                .build();
    }

    public TaskResponse createTaskResponse() {
        return TaskResponse.builder()
                .id(id)
                .title(title)
                .description(description)
                .status(status)
                .priority(priority)
                .build();
    }
}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/context/TaskTestContext.kt
package com.example.demo.e2e.context

import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus
import com.example.demo.e2e.data.TaskData
import com.example.demo.e2e.data.TaskResponse
import java.util.UUID

data class TaskTestContext(
    val id: String = UUID.randomUUID().toString(),
    val title: String = randomAlphabetic(12),
    val description: String = randomAlphabetic(12),
    val status: TaskStatus = TaskStatus.TODO,
    val priority: TaskPriority = TaskPriority.MEDIUM,
) {
    fun createTaskData(): TaskData = TaskData(
        title = title,
        description = description,
        status = status,
        priority = priority,
    )

    fun createTaskResponse(): TaskResponse = TaskResponse(
        id = id,
        title = title,
        description = description,
        status = status,
        priority = priority,
    )

    private companion object {
        fun randomAlphabetic(length: Int): String = buildString(length) {
            repeat(length) {
                append(('A'..'Z').random())
            }
        }
    }
}
